queue/redis/opts.go

```diff
@@ -69,3 +71,75 @@ func WithTimeout(timeout time.Duration) ClientOpt {
 		return nil
 	}
 }
+
+// WithPrivateKey sets the private key in the queue client for Redis.
+//
+//nolint:dupl // ignore similar code
+func WithPrivateKey(privateKeyEncoded string) ClientOpt {
+	return func(c *client) error {
+		c.Logger.Trace("configuring private key in redis queue client")
+
+		if len(privateKeyEncoded) == 0 {
+			c.Logger.Warn("unable to base64 decode private key, provided key is empty. queue service will be unable to sign items")
+			return nil
+		}
+
+		privateKeyDecoded, err := base64.StdEncoding.DecodeString(privateKeyEncoded)
```

Comment:
good question, i used base64 here because the private/public keys i generated arent in unicode.
(using this process to make it easier: https://go.dev/play/p/-go_7SnJbnP)
base64 encoding them made it easy to copy+paste, store elsewhere, put into transit, easier handling. 

for the other values in the server? id argue we should base64 anything that isnt in a format that is easy to handle.
though i think the other keys are fine because we recommend generating them using `-hex` which isnt a valid way to generate the keypair for queue signing.
```
openssl rand -hex 16
```
from docs
https://go-vela.github.io/docs/installation/server/docker/#step-4-create-the-private-key

so... honestly yeah we should base64 more, but for queue signing i believe its more important

Reply:
Cool, thanks for the explanation